### PR TITLE
Align the boolean validity claim

### PR DESCRIPTION
**Pull request by marinaado**

### CODE REVIEW

File: draft-demarco-oauth-status-assertions.md [hunk 1]
```diff
@@ -553,7 +553,7 @@ where the format is JWT.
     "exp": 1504785536,
     "credential_hash": $hash-about-Issuer-Signed-JWT,
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": true,
+    "credential_status_validity": valid,
```
Comment (OR13) [inline draft-demarco-oauth-status-assertions.md:556]:
```suggestion
    "credential_status_validity": "valid",
```

File: draft-demarco-oauth-status-assertions.md [hunk 2]
```diff
@@ -685,10 +685,10 @@ variety of scenarios.
 
 # Detailed Status Assertions
 
-Status Assertions can introduce a more accurate level of detail, and therefore not necessarly limited to simple boolean information.
-This enables Verifier policies to be conditioned on the presence of secured information, instead of the absence of information.
+Status Assertions can introduce a more accurate level of detail about the credential status.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:688]:
```suggestion
Status Assertions can introduce a more accurate level of detail about the Credential status.
```

File: draft-demarco-oauth-status-assertions.md [hunk 3]
```diff
@@ -739,9 +738,9 @@ An example of an enumeration status is:
     "exp": 1504785536,
     "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": false,
+    "credential_status_validity": invalid,
     "credential_status_detail": {
-      "state": "suspended", // or "revoked"
+      "state": "Document lost"
```
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:743]:
```suggestion
      "state": "document lost"
```

File: draft-demarco-oauth-status-assertions.md [hunk 4]
```diff
@@ -576,7 +576,7 @@ The Status Assertion MUST contain the parameters defined below.
 | **exp** | UNIX Timestamp with the expiration time of the JWT. It MUST be greater than the value set for `iat`. | {{RFC9126}}, {{RFC7519}}, {{RFC7515}} |
 | **credential_hash** | Hash value of the Digital Credential the Status Assertion is bound to. | this specification |
 | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential to which the Status Assertion is bound. The value SHOULD be set to `sha-256`. | this specification |
-| **credential_status_validity**| Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. This parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
+| **credential_status_validity**| String value indicating the absolute validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. Status validity MUST be `valid`, `invalid` or `suspended`. Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```
Comment (OR13) [inline draft-demarco-oauth-status-assertions.md:579]:
```suggestion
| **credential_status_validity**| String value indicating the absolute validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```

File: draft-demarco-oauth-status-assertions.md [hunk 5]
```diff
@@ -707,10 +707,9 @@ An example of a boolean status is:
     "exp": 1504785536,
     "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": false,
+    "credential_status_validity": suspended,
```
Comment (OR13) [inline draft-demarco-oauth-status-assertions.md:710]:
```suggestion
    "credential_status_validity": "suspended",
```

File: draft-demarco-oauth-status-assertions.md [hunk 6]
```diff
@@ -739,9 +738,9 @@ An example of an enumeration status is:
     "exp": 1504785536,
     "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": false,
+    "credential_status_validity": invalid,
```
Comment (OR13) [inline draft-demarco-oauth-status-assertions.md:741]:
```suggestion
    "credential_status_validity": "invalid",
```

File: draft-demarco-oauth-status-assertions.md [hunk 7]
```diff
@@ -770,7 +769,7 @@ An example of dynamic status using a small matrix:
     "exp": 1504785536,
     "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": true,
+    "credential_status_validity": valid,
```
Comment (OR13) [inline draft-demarco-oauth-status-assertions.md:772]:
```suggestion
    "credential_status_validity": "valid",
```

File: draft-demarco-oauth-status-assertions.md (hunk 5 above)
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:710]:
```suggestion
    "credential_status_validity": "suspended",
```

File: draft-demarco-oauth-status-assertions.md (hunk 6 above)
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:741]:
```suggestion
    "credential_status_validity": "invalid",
```

File: draft-demarco-oauth-status-assertions.md (hunk 7 above)
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:772]:
```suggestion
    "credential_status_validity": "valid",
```

**Comment by marinaado** [comment]
I am concerned if the examples given for the credential_status_detail parameter are good to explain its use. The examples we give don't provide any additional information then what is already in the credential_status_validity parameter.

File: draft-demarco-oauth-status-assertions.md [hunk 8]
```diff
@@ -685,14 +685,14 @@ variety of scenarios.
...
+Status Assertions can introduce a more accurate level of detail about the Credential status.
+This enables Verifier policies to be conditioned on the presence of authorative information.
 This section proposes syntax to support detailed assertions.
-The `credential_status_validity` claim MUST be present and be either `true` or `false`.
+The `credential_status_validity` claim MUST be present.
 The `credential_status_detail` claim MAY be present and if present MUST be an object.
 The semantics of the claims within the `credential_status_detail` object are determined by the Credential Issuer.
 
-An example of a boolean status is:
+An example of bolean detail status is:
```
Comment (fmarino-ipzs) [inline draft-demarco-oauth-status-assertions.md:695]:
```suggestion
An example of boolean detail status is:
```

File: draft-demarco-oauth-status-assertions.md [hunk 9]
```diff
@@ -576,7 +576,7 @@ The Status Assertion MUST contain the parameters defined below.
 | **exp** | UNIX Timestamp with the expiration time of the JWT. It MUST be greater than the value set for `iat`. | {{RFC9126}}, {{RFC7519}}, {{RFC7515}} |
 | **credential_hash** | Hash value of the Digital Credential the Status Assertion is bound to. | this specification |
 | **credential_hash_alg** | The Algorithm used of hashing the Digital Credential to which the Status Assertion is bound. The value SHOULD be set to `sha-256`. | this specification |
-| **credential_status_validity**| Boolean value indicating the absolute validity of the Credential linked to the Status Assertion. This parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
+| **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```
Comment (fmarino-ipzs) [inline draft-demarco-oauth-status-assertions.md:579]:
```suggestion
| **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry (TBD).Values from 0 to 2 are defined by this specification with the following meaning: 0-VALID, 1-INVALID, 2-SUSPENDED. Credential Issuers MAY extend with other numeric values based on use cases of interest. Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```

**Review by fmarino-ipzs — CHANGES_REQUESTED** [review]

File: draft-demarco-oauth-status-assertions.md [hunk 10]
```diff
@@ -596,7 +596,11 @@ the endpoint where the Wallet Instances can request Status Assertions.
 the Wallet Instance to hash the Digital Credential for which the
 Status Assertion is requested,  using one of the hash algorithms listed
 in the [IANA-HASH-REG].
+- `credential_status_detail_supported`. OPTIONAL. JSON array that outlines the details of each Credential's validity status supported by the Credential Issuer. This parameter MAY be used to extend the values defined in Section [Status Assertion](#status-assertion). Each entry MUST contain the following parameters:
 
+    - `credential_status_validity`. Numerical value indicating the validity of the Credential.  
+    - `state`. String value of a Credential status supported.
+    - `description`. String containing the human-readable description of the status related to this object.
```
Comment (fmarino-ipzs) [inline draft-demarco-oauth-status-assertions.md:603]: @marinaado , I added this additional optional parameter in the metadata of a CI. The idea is that a CI MAY use this parameter to extend the values defined within this specification (0, 1, 2) and the Wallet can obtain from the CI's Metadata the detailed info to prompt the User.  
WDYT about this? 

@OR13 , @peppelinux  ^^

File: draft-demarco-oauth-status-assertions.md (hunk 9 above)
Comment (marinaado) [reply draft-demarco-oauth-status-assertions.md:579]: I think that we agreed that we don't want to hardcode the values in this spec, as we want to align values with related parameter in Status List specification. We assume that they will register the values in the IANA registry so we want to reference to that instead. I think thought that we can use this values in the examples.

Comment (OR13) [inline draft-demarco-oauth-status-assertions.md:579]:
```suggestion
| **credential_status_validity**| Numerical value indicating the validity of the Credential linked to the Status Assertion, describing its state, mode, condition or stage. The value MUST be from the IANA registry ( as described in Section 7.1 of draft-ietf-oauth-status-list). Status validity parameter is REQUIRED, and the Verifier MUST verify its presence and value to assess the Credential's validity. | this specification |
```

Comment (OR13) [reply draft-demarco-oauth-status-assertions.md:579]: This will need adjustment to link up properly as a well formed reference

Comment (fmarino-ipzs) [reply draft-demarco-oauth-status-assertions.md:579]: Even if we will have the values registered in IANA, why shouldn't we have in this specification? IMHO having them explicitly here could be useful for implementers that are already implementing this. Am I missing something?

Comment (OR13) [reply draft-demarco-oauth-status-assertions.md:579]: I created this issue, https://github.com/peppelinux/draft-demarco-oauth-status-assertions/issues/90

and am resolving this conversation.

Comment (OR13) [reply draft-demarco-oauth-status-assertions.md:579]: I also cross filed this issue regarding the discussion on reserved for private use range here: https://github.com/oauth-wg/draft-ietf-oauth-status-list/issues/181

File: draft-demarco-oauth-status-assertions.md [hunk 11]
```diff
@@ -707,7 +711,7 @@ An example of a boolean status is:
     "exp": 1504785536,
     "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": false,
+    "credential_status_validity": 3,
     "credential_status_detail": {
       "revoked": false,
       "suspended": true,
```
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:717]:
```suggestion
      ...
```

File: draft-demarco-oauth-status-assertions.md [hunk 12]
```diff
@@ -739,9 +743,9 @@ An example of an enumeration status is:
     "exp": 1504785536,
     "credential_hash": "xnlAq6Ma8fgu1z4hdGphJnKLulaVHpLCFeZFUGpQ2dA",
     "credential_hash_alg": "sha-256",
-    "credential_status_validity": false,
+    "credential_status_validity": 3,
     "credential_status_detail": {
-      "state": "suspended", // or "revoked"
+      "state": "suspended",
```
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:748]:
```suggestion
      ...
```

**Comment by OR13** [comment]
I filed this issue related to i18n and the examples. https://github.com/peppelinux/draft-demarco-oauth-status-assertions/issues/91

File: draft-demarco-oauth-status-assertions.md (hunk 10 above)
Comment (marinaado) [reply draft-demarco-oauth-status-assertions.md:603]: Agreed that the IANA registry reserved range for private use should be used to communicate values outside of the ones defined in the IANA registry.

**Review by peppelinux — APPROVED** [review]

File: draft-demarco-oauth-status-assertions.md [hunk 13]
```diff
@@ -596,7 +596,11 @@ the endpoint where the Wallet Instances can request Status Assertions.
 the Wallet Instance to hash the Digital Credential for which the
 Status Assertion is requested,  using one of the hash algorithms listed
 in the [IANA-HASH-REG].
+- `credential_status_detail_supported`. OPTIONAL. JSON array that outlines the details of each Credential's validity status supported by the Credential Issuer. This parameter MAY be used to extend the values defined in Section [Status Assertion](#status-assertion). Each entry MUST contain the following parameters:
 
+    - `credential_status_validity`. Numerical value indicating the validity of the Credential.
```
Comment (peppelinux) [inline draft-demarco-oauth-status-assertions.md:601]:
```suggestion
    - `credential_status_validity`. Numerical value indicating the validity of the Credential.
```

**Review by OR13 — APPROVED** [review]

**Review by fmarino-ipzs — APPROVED** [review]

**Pull request merged**